dts/bindings/clock/st,stm32-hse-clock.yaml

```diff
@@ -13,3 +13,14 @@ properties:
     description: |
       HSE crystal oscillator bypass
       Set to the property to by-pass the oscillator with an external clock.
+
+  css-enabled:
+    type: boolean
+    description: |
+      HSE clock security system enabled
+      Set the property to enable the clock security system for the HSE clock.
+
+      If a failure is detected on the HSE clock, the HSE oscillator is automatically disabled,
+      a clock failure event is sent to timers, and a non-maskable interrupt is generated to
+      inform the software about the failure, allowing the MCU to perform rescue operations.
```

Comment:
The easiest way is to set `CONFIG_RUNTIME_NMI=y` and call `z_arm_nmi_set_handler(HAL_RCC_NMI_IRQHandler)`. The HAL NMI handler deals with clearing the interrupt and calls `HAL_RCC_CSSCallback` (the default implementation does nothing).

Since the default NMI handler just loops forever the handler should probably be set before enabling CSS. So should that be done automatically with `css-enabled`?

Then I could just add a note in the binding file saying that you need to implement your own `HAL_RCC_CSSCallback`.

Reply:
Thanks for the details.
What about integrating this in the STM32 driver itself in a zephyr way ?
`RUNTIME_NMI` could be enabled based on hse-css value (using `dt_nodelabel_has_prop` macro), then you could re-implement HAL_RCC_NMI_IRQHandler (using LL Apis, because this is the LL API used in the clock control driver), then you would call a weak callback (available in the driver, but doing nothing), similarly to what is done in STM32Cube package.

Reply:
> Is it necessary to reimplement HAL_RCC_NMI_IRQHandler? 

This is not mandatory. We're trying to maximize use of LL when possible. In general LL is preferred due to lower footprint. Then for consistency, we try to minimize use of HAL for consistency when both are equivalent.
So if this is working using LL API w/o too much trouble, please use them. If this is causing issues because LL API miss cross series consistency and requires using #ifdef ... let's keep the HAL_RCC call

Reply:
I've pushed an update using the HAL API for handling the HSE CSS interrupt.

The LL API doesn't provide much consistency across different devices. There are different versions of the handler even within the "common" ones for C0 and G0 devices. To implement this with only LL API we'd need to copy all the various implementations of `HAL_RCC_NMI_IRQHandler` and have a lot of ugly #ifdef'ing. And since the HAL is always used for stm32_clock_control it doesn't seem worthwhile.